src/browser/BrowserService.cpp

```diff
@@ -925,60 +928,75 @@ int BrowserService::sortPriority(const Entry* entry,
                                  const QString& baseSubmitUrl,
                                  const QString& fullUrl) const
```

Comment:
I don't like the parameters of this function. We should be passing a list of URL's (not the entry itself) and the desired match URL only. Slicing and dicing the desired match URL should happen within the function. The function basically relies on the caller to do the right thing to make this function work properly.

Reply:
True. Passing the list could also include the Additional URL's.

Reply:
Pushed the fix. `submitUrl` needed to be passed also because we make comparisons to that too.